### PR TITLE
Extend fqcn rule to complain on `collections` keyword

### PR DESCRIPTION
**Pull request by jcgruenhage**
I had opened a discussion about this a month ago (https://github.com/ansible/ansible-lint/discussions/2728), but since there's no interaction happening there I'm opening a PR.

### CODE REVIEW

**Comment by cidrblock** [comment]
This is _awesome_. TY

**Review by cidrblock — APPROVED** [review]
A test would be cool here

File: src/ansiblelint/rules/fqcn.py [hunk 1]
```diff
@@ -159,6 +160,20 @@ def matchtask(
                     )
         return result
 
+    def matchplay(self, file: Lintable, data: dict[str, Any]) -> list[MatchError]:
+        if file.kind != "playbook":
+            return []
+        if "collections" in data:
+            return [
+                self.create_matcherror(
+                    message="Don't use collections keyword.",
```
[the remark this answers is not in the document]
Comment (jcgruenhage) [reply src/ansiblelint/rules/fqcn.py:169]: Sure, can do.

**Comment by jcgruenhage** [comment]
@cidrblock thanks for the feedback. I've changed the message based on it and extended the test case to also cover this.

**Review by ssbarnea — APPROVED** [review]

**Comment by jcgruenhage** [comment]
Rebased on top of main to re-run CI. The failures weren't related to the PR, just timeouts when talking to galaxy.ansible.com. Would it be possible to add some retries to those calls, so that a single timeout doesn't fail a whole CI job?

**Comment by ssbarnea** [comment]
@jcgruenhage I know that galaxy unreliability is a PITA and I already did some work to reduce its use, still I was not able to fully de-couple it, especially as it is impossible to install roles offline.

**Comment by jcgruenhage** [comment]
Well, but CI passed this time, so it's all good I suppose :D

For merging, it's missing those two pending reviews I'd assume?

**Review by ganeshrn — APPROVED** [review]

**Pull request merged**